src/Workspaces/CSharp/Portable/Extensions/ITypeSymbolExtensions.TypeSyntaxGeneratorVisitor.cs

```diff
@@ -62,25 +62,51 @@ public override TypeSyntax VisitArrayType(IArrayTypeSymbol symbol)
             {
                 ThrowIfNameOnly();
 
-                var underlyingNonArrayType = symbol.ElementType;
-                while (underlyingNonArrayType.Kind == SymbolKind.ArrayType)
+                ITypeSymbol underlyingType = symbol;
+
+                while (underlyingType is IArrayTypeSymbol innerArray)
                 {
-                    underlyingNonArrayType = ((IArrayTypeSymbol)underlyingNonArrayType).ElementType;
+                    underlyingType = innerArray.GetElementTypeWithAnnotatedNullability();
+
+                    if (underlyingType.GetNullability() == NullableAnnotation.Annotated)
+                    {
+                        // If the inner array we just moved to is also nullable, then
```

Comment:
```suggestion
                        // 🐉 If the inner array we just moved to is also nullable, then
```

Reply:
What does the dragon mean?

Reply:
Won't fix.

Reply:
(I storngly disagree with use of emoji in general as demonstrated by @genlu: those can be confused as the meaning. Text less so.)

Reply:
It's a fairly common term in programming sourced originally from https://en.wikipedia.org/wiki/Here_be_dragons